### PR TITLE
add docs for equations

### PR DESCRIPTION
**Pull request by simonbyrne**
Start on #287.

### CODE REVIEW

File: docs/src/equations.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+!!! note
+
+    This follows what is _currently_ implemented in `examples`: it should be kept up-to-date as code is modified. If you think something _should_ be changed (but hasn't been), please add a note.
+
+
+## Prognostic variables
+
+* ``\rho``: _mass_ in kg/m³. This is discretized at cell centers.
+* ``\boldsymbol{u}`` _velocity_, a vector in m/s. This is discretized via ``\boldsymbol{u} = \boldsymbol{u}_v + \boldsymbol{u}_h`` where
```
Comment (szy21) [inline docs/src/equations.md:12]:
```suggestion
* ``\boldsymbol{u}`` _velocity_, a vector in m/s. This is discretized via ``\boldsymbol{u} = \boldsymbol{u}_h + \boldsymbol{u}_v`` where
```

File: docs/src/equations.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+We make use of the following operators
+
+### Reconstruction
+
+* ``I^c`` is the [face-to-center reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateF2C)
+* ``I^f`` is the [center-to-face reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateC2F)
+  - Currently this is just the arithmetic mean, but we will need to use a weighted version with stretched vertical grids.
+* ``U^f`` is the [center-to-face upwind product operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)
+  - Currently this is first-order, we will switch this to 3rd order.
```
Comment (szy21) [inline docs/src/equations.md:30]:
```suggestion
  - Currently this is first-order, we will switch this to third-order.
```
(Sorry about my OCD here, you can ignore it:)

**Review by bischtob — APPROVED** [review]
Good updates!

File: docs/src/equations.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+- ``C_h`` is the [horizontal curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Curl)
+  - ``C_h[\boldsymbol{u}_h]`` returns a vector with only a vertical _contravariant_ component.
+  - ``C_h[\boldsymbol{u}_v]`` returns a vector with only a horizontal _contravariant_ component.
+- ``C^f_v`` is the [center-to-face vertical curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.CurlC2F)
+  - ``C^f_v[\boldsymbol{u}_h]`` returns a vector with only a horizontal _contravariant_ component.
+  - the boundary is defined to be zero.
+
+## Auxiliary and derived quantities
+
+* ``\boldsymbol{\Omega}`` is the planetary angular velocity (Coriolis). We currently use a shallow-atmosphere approximation, with
```
Comment (szy21) [inline docs/src/equations.md:49]:
```suggestion
* ``\boldsymbol{\Omega}`` is the planetary angular velocity. We currently use a shallow-atmosphere approximation, with
```

File: docs/src/equations.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  - The fluxes at the surface and top are set to zero.
+- ``G_h`` is the [discrete horizontal spectral gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Gradient)
+- ``G^f_v`` is the [center-to-face vertical gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.GradientC2F)
+  - the gradient is set to 0 at the boundary.
+- ``C_h`` is the [horizontal curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Curl)
+  - ``C_h[\boldsymbol{u}_h]`` returns a vector with only a vertical _contravariant_ component.
+  - ``C_h[\boldsymbol{u}_v]`` returns a vector with only a horizontal _contravariant_ component.
+- ``C^f_v`` is the [center-to-face vertical curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.CurlC2F)
+  - ``C^f_v[\boldsymbol{u}_h]`` returns a vector with only a horizontal _contravariant_ component.
+  - the boundary is defined to be zero.
```
Comment (szy21) [inline docs/src/equations.md:45]:
```suggestion
  - the curl is set to 0 at the top and bottom boundaries.
```

File: docs/src/equations.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+* ``I^f`` is the [center-to-face reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateC2F)
+  - Currently this is just the arithmetic mean, but we will need to use a weighted version with stretched vertical grids.
+* ``U^f`` is the [center-to-face upwind product operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)
+  - Currently this is first-order, we will switch this to 3rd order.
+
+### Differentiation operators
+
+- ``D_h`` is the [discrete horizontal spectral divergence](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Divergence)
+- ``D^c_v`` is the [face-to-center vertical divergence](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.DivergenceF2C)
+  - The fluxes at the surface and top are set to zero.
```
Comment (szy21) [inline docs/src/equations.md:36]:
```suggestion
  - The fluxes are set to zero at the top and bottom boundaries.
```

File: docs/src/equations.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+- ``G_h`` is the [discrete horizontal spectral gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Gradient)
+- ``G^f_v`` is the [center-to-face vertical gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.GradientC2F)
+  - the gradient is set to 0 at the boundary.
```
Comment (szy21) [inline docs/src/equations.md:39]:
```suggestion
  - the gradient is set to 0 at the top and bottom boundaries.
```

**Review by szy21 — APPROVED** [review]
Thanks Simon!

**Pull request merged**